### ISSUE TITLE
[BUG] HTML representation of estimator parameters do not show in notebooks

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#5924]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating system

- [x] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

For example one of the following:
- Linux Ubuntu 22.04

### Python version

- [ ] 3.14
- [x] 3.13
- [ ] 3.12
- [ ] 3.11
- [ ] 3.10

### nilearn version

dev

### Expected behavior

since sklearn 1.8 estimators can have an improved HTML representation

https://scikit-learn.org/stable/auto_examples/release_highlights/plot_release_highlights_1_8_0.html#html-representation-of-estimators

I would expect that the Nilearn estimators should be able to take advantage of this but it does not see to be the case

### Current behavior & error messages

This is what I got (not visible on the image but the parameters show up their documentation when hovering over them).

<img width="1646" height="1324" alt="Image" src="https://github.com/user-attachments/assets/844e95be-ffab-4f64-82ed-8a33465a32d5" />

### Steps and code to reproduce bug

tried with this marimo notebook

```python
import marimo

__generated_with = "0.18.4"
app = marimo.App(width="full")


@app.cell
def _():
    import numpy as np
    import nibabel as nib
    from nilearn.decoding import Decoder

    rng = np.random.RandomState(42)
    X = rng.rand(5, 5, 5, 20)
    X_img = nib.Nifti1Image(X, np.eye(4))
    y = np.array([0] * 10 + [1] * 10)

    decoder = Decoder()
    decoder.fit(X_img, y)
    return


@app.cell
def _():
    from sklearn import linear_model

    reg = linear_model.LinearRegression()

    reg.fit([[0, 0], [1, 1], [2, 2]], [0, 1, 2])
    return


if __name__ == "__main__":
    app.run()
```

**Comment by Remi-Gau** [comment]
This makes me realize that we could maybe make our life easier in our reports by reusing the sklearn way to display the estimator parameters.

Note that this is how we do it.

<img width="815" height="622" alt="Image" src="https://github.com/user-attachments/assets/8809693f-bd9c-4e94-9c5b-7958dfa082d9" />
